### PR TITLE
Update and re-lock dependencies; Reintroduce Codecov

### PR DESCRIPTION
**Pull request by KPrasch**
Includes hendrix v3.2.3 in response to Issue #909 ; Derived from commits in #825, removes coveralls and reintroduces codecov GH integration via circleCI orb. Also opens follow-up Issue: <https://github.com/nucypher/nucypher/issues/911>

Implements codecov orb: <https://circleci.com/blog/making-code-coverage-easy-to-see-with-the-codecov-orb/>
Applies codecov configuration / usage fixes documented here: <https://pytest-cov.readthedocs.io/en/latest/reporting.html>
see codecov results here: <https://codecov.io/gh/nucypher/nucypher>

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ maya = "*"
 flask = "*"
 flask_sqlalchemy = "*"
 # Third-Party Ethereum
-py-evm = "==0.2.0a39"
+py-evm = ">=0.2.0a39"
```
Comment (michwill) [inline Pipfile]: Does it really work with this newer version? I think, there were issues with `eth-tester`?

Comment (KPrasch) [reply Pipfile]: I think it remained locked at `0.2.0a39`

Comment (michwill) [reply Pipfile]: Sure... But here in Pipfile it's updated, and then `--skip-lock` will install 0.2.0a42 or whatever, and eth-tester will fail?
Unless they shipped a newer version

Comment (KPrasch) [reply Pipfile]: I think with the presence of a `Pipfile.lock` - there is no need to use `--skip-lock` and the best way to install is `pipenv sync`, although I think `pipenv install` will have a similar effect

Comment (michwill) [reply Pipfile]: Well, at my machine, it tries to relock for some reason, and that happens forever.
Why not keep it ==0.2.0a39 as long as it is not known to work at higher versions?

Comment (michwill) [reply Pipfile]: (until eth-tester gets updated)

Comment (michwill) [reply Pipfile]: upd: Pipfile.lock works indeed. But still, needs to be workable even with --skip-lock

Comment (KPrasch) [reply Pipfile]: Reverted - :heavy_check_mark:

**Comment by KPrasch** [comment]
Also waiting on <https://circleci.com/workflow-run/3caf0197-fcb5-4061-891b-d35b09386736>

**Pull request merged**